src/common/drpc.c

```diff
@@ -64,29 +64,36 @@ daos_drpc_free(void *allocater_data, void *pointer)
  * \param	ctx	Active dRPC context
  * \param	module	Module ID for the new call
  * \param	method	Method ID for the new call
+ * \param	callp	Newly allocated Drpc__Call
  *
- * \return	Newly allocated Drpc__Call, or NULL if it couldn't be allocated
+ * \returns	On success returns 0 otherwise returns negative error condition.
  */
-Drpc__Call *
-drpc_call_create(struct drpc *ctx, int32_t module, int32_t method)
+int
+drpc_call_create(struct drpc *ctx, int32_t module, int32_t method,
+		 Drpc__Call **callp)
 {
 	Drpc__Call *call;
 
-	if (ctx == NULL) {
+	if (callp)
+		*callp = NULL;
```

Comment:
minor nit - but if you move this after the null checks below, the if check won't be necessary.

Reply:
True, but one of the tests segfaults if I do that.